typescript-for-haskellers/article.md

```diff
@@ -0,0 +1,726 @@
+
+Moreover, UI and UX standards are increasing.
+People like simple interfaces with enjoyable.
+Application with worse functionality and better UI/UX may win on the market against the other one with bad design.
+
+One might say that doesn't matter what your application is about or what part of it you write.
+In the end in most of the cases you will need to give access for it to smb.
+Whether it is an external user or your colleague using internal products.
+
+So, in Serokell we use TypeScript for writing web pages.
+And as our main language is Haskell, in this article we want to explain how knowledge from it will help you to write TypeScript code.
+
+## Why TypeScript?
+
+Basically our Web world consists of 3 things: HTML, CSS, and JavaScript.
+The first one is about content, the second one about this content style and the last one about how you can interact with this content.
+Moreover, these 3 things are just files, and you need a way to read and show them.
+That is what browsers do.
+
+Historically this set of files was perfect to write web pages.
+And because of the enormous growth of the web JavaScript in particular has grown enormous as well.
+
+So, as a result any new language which allow you to write web applications translate its code to JavaScript.
+And there are a lot of such languages and language extensions: TypeScript, Kotlin JS, CoffeeScript, Scala JS, Babel, ClojureScript, etc.
+Generally you can write web applications almost on every language.
+
+As we like Haskell, maybe we can write our Web application on functional languages or even on Haskell? And the answer is yes.
+There is strongly-typed PureScript with Haskell like syntax, purely functional Elm, Reflex framework for Haskell with GHCJS, F# with Fabel.
+
+But when we use Haskell we know that people like it and customers want it, and we can't say the same thing about technologies above.
+Customers will be locked in this solution as most likely they will not be able to maintenance such application by themselves.
+On our side we will be locked in using JS libraries which number is enormous, build problems, and fast team expanding.
+
+Thus, we needed a language which will easy support all modern technologies from web development, big community, and give us a possibility to use our knowledge from Haskell.
+And here it is -- TypeScript.
+It is static typing language which popularity grows from year to year.
+A lot of new libraries are writen on it, a lot of have type declarations, and it provides an easy way to use plain JS libraries as well.
+
+Now it is time to see how your Haskell experience may help you in learning TypeScript and how you can improve your code with it.
+
+In this article we don't have a goal to cover all similarities or differences.
+We also will not say about disadvantages of TypeScript type system against Haskell or how it works under the hood.
+Our goal is to show what and how you can use types in TypeScript.
+
+## Similar concepts
```

Comment:
Since we have a section about similar concepts, perhaps there are some unsimilar pitfalls that Haskell developers might fall into?

Reply:
Hm, I think I should just somehow rename the section as all article is about `Similar concepts` but this part is just about easiest things

> It would be a nice "mirror section" for this one

I didn't want to mention them because in this case it will become a TypeScript guide but not a brief introduction to types in it. I also didn't want to describe all code and how it work, only examples, so that reader may go and google it and find more examples.